core/src/main/java/org/elasticsearch/snapshots/SnapshotId.java

```diff
@@ -115,4 +120,35 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeString(uuid);
     }
 
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field(NAME, name);
+        builder.field(UUID, uuid);
+        builder.endObject();
+        return builder;
+    }
+
+    public static SnapshotId fromXContent(XContentParser parser) throws IOException {
+        // the new format from 5.0 which contains the snapshot name and uuid
+        if (parser.currentToken() == XContentParser.Token.START_OBJECT) {
+            String name = null;
+            String uuid = null;
+            while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+                String currentFieldName = parser.currentName();
+                parser.nextToken();
+                if (NAME.equals(currentFieldName)) {
+                    name = parser.text();
+                } else if (UUID.equals(currentFieldName)) {
+                    uuid = parser.text();
+                }
+            }
+            return new SnapshotId(name, uuid);
+        }
+        // the old format pre 5.0 that only contains the snapshot name, use the name as the uuid too
+        else {
+            return new SnapshotId(parser.text(), SnapshotId.UNASSIGNED_UUID);
```

Comment:
You're right, the UNASSIGNED_UUID was there before when a combination of name and uuid was used for the blobId.  Now that its just uuid, the UNASSIGNED_UUID can go away.  Thanks for catching that!